src/nvim/CMakeLists.txt

```diff
@@ -816,6 +816,14 @@ if(EXISTS ${DEPS_PREFIX}/lib/nvim/parser)
     COMMAND ${CMAKE_COMMAND} -E ${COPY_DIRECTORY} ${DEPS_PREFIX}/lib/nvim/parser ${BINARY_LIB_DIR}/parser)
 endif()
 
+if(CYGWIN AND (EXISTS ${DEPS_PREFIX}/bin/cyglua51.dll))
```

Comment:
Kinda hacky, I think the "correct" way to do this would be to search for `cyglua51.dll` in `cmake/FindLuajit.cmake` and drop something like `list(APPEND LUAJIT_NAMES cyglua51)` (that or maybe it's `libcyglua511`).